src/PagerViewNative.tsx

```diff
@@ -1,17 +1,37 @@
-import type { ReactElement } from 'react';
-import { HostComponent, requireNativeComponent, UIManager } from 'react-native';
-import type { PagerViewProps } from './types';
+import {
+  GestureResponderEvent,
+  requireNativeComponent,
+  StyleProp,
+  ViewStyle,
+} from 'react-native';
+import type {
+  Orientation,
+  PagerViewOnPageScrollEvent,
+  PagerViewOnPageSelectedEvent,
+  PageScrollStateChangedNativeEvent,
+  TransitionStyle,
+} from './types';
 
 const VIEW_MANAGER_NAME = 'RNCViewPager';
 
-interface PagerViewViewManagerType extends HostComponent<PagerViewProps> {
-  getInnerViewNode(): ReactElement;
-}
+type PagerViewNativeProps = {
```

Comment:
Why old implementation has been removed ? we don't need to copy/paste the same types

Reply:
The native component and js component take different props.  Native component requires `childrenKeys`, `count`, `offset`, and `onMoveShouldSetResponderCapture`, which js component does not accept.  Js component accepts `initialPage`, `keyboardDismissMode`, `data`, `keyExtractor`, `maxRenderWindow`, and `renderItem`, which native component does not accept.

I could factor out the common props into a new type to inherit from, to avoid duplication.